services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/kafka/KafkaBasedCommandConsumerFactoryImpl.java

```diff
@@ -108,6 +113,9 @@ public KafkaBasedCommandConsumerFactoryImpl(
         this.metrics = Objects.requireNonNull(metrics);
         this.tracer = Objects.requireNonNull(tracer);
 
+        kafkaClientMetricsSupport = metrics instanceof MicrometerBasedMetrics ? new MicrometerKafkaClientMetricsSupport(((MicrometerBasedMetrics) metrics).getRegistry()) : null;
+        Optional.ofNullable(kafkaClientMetricsSupport).ifPresent(ms -> kafkaProducerFactory.setMetricsSupport(ms));
+
```

Comment:
I've now added a `KafkaClientMetricsSupport` parameter.

> Alternatively, CommandRouterMetrics could also extend KafkaClientMetricsSupport, couldn't it?

Then it would make sense to also let `AmqpAdapterMetrics` extend `KafkaClientMetricsSupport` for example. But that would require Kafka dependencies to be added to the various adapter base modules.
I'd rather keep handling of Hono message metrics and Kafka clients metrics in separate classes then.